river/tree/streaming_gradient_tree.py

```diff
@@ -0,0 +1,460 @@
+            else None
+        )
+
+        if F is None:
+            return 1.0
+
+        return 1 - FTest.cdf(F, 1, n_observations - 1)
+
+    def learn_one(self, x, y, *, w=1.0):
+        self._update_features_stats(x)
+
+        raw_pred = self._raw_prediction(x)
+        label = self._target_transform(y)
+
+        grad_hess = self._objective.compute_derivatives(label, raw_pred)  # noqa
+
+        # Update the tree with the gradient/hessian info
+        self._update_tree(x, grad_hess, w)
+
+        return self
+
+    def _raw_prediction(self, x):
+        """ Obtain a raw prediction for a single instance. """
+
+        pred = self._root.sort_instance(x).leaf_prediction()
+        return self._objective.transfer(pred)  # noqa
+
+    @property
+    def n_nodes(self):
+        n = 0
+        to_visit = [self._root]
+        while len(to_visit) > 0:
+            node = to_visit.pop(0)
+            n += 1
+
+            if node.children is not None:
+                for child in node.children.values():
+                    to_visit.append(child)
+        return n
+
+    @property
+    def n_splits(self):
+        return self._n_splits
+
+    @property
+    def n_node_updates(self):
+        return self._n_node_updates
+
+    @property
+    def n_observations(self):
+        return self._n_observations
+
+    @property
+    def depth(self):
+        return self._depth
+
+
+class StreamingGradientTreeClassifier(BaseStreamingGradientTree, base.Classifier):
+    """Streaming Gradient Tree for classification.
+
+    Binary decision tree classifier that minimizes the binary cross-entropy to guide its growth.
+
+    Streaming Gradient Trees (SGT) directly minimize a loss function to guide tree growth and
+    update their predictions. Thus, they differ from other incrementally tree learners that do
+    not directly optimize the loss, but a data impurity-related heuristic.
+
+    Parameters
+    ----------
+    delta
+        Define the significance level of the F-tests performed to decide upon creating splits
+        or updating predictions.
+    grace_period
+        Interval between split attempts or prediction updates.
+    init_pred
+        Initial value predicted by the tree.
+    max_depth
+        The maximum depth the tree might reach. If set to `None`, the trees will grow
+        indefinitely.
+    lambda_value
+        Positive float value used to impose a penalty over the tree's predictions and force
+        them to become smaller. The greater the lambda value, the more constrained are the
+        predictions.
+    gamma
+        Positive float value used to impose a penalty over the tree's splits and force them to
+        be avoided when possible. The greater the gamma value, the smaller the chance of a
+        split occurring.
+    nominal_attributes
+        List with identifiers of the nominal attributes. If None, all features containing
+        numbers are assumed to be numeric.
+    quantization_strategy
```

Comment:
What about using a tuple for this parameter? They are non-mutable and by pairing `(strategy, value)` we can avoid potential errors when setting these parameters.

Reply:
Changed in commit [7b44573](https://github.com/online-ml/river/pull/458/commits/7b445738c7d9f730f50aeddc546fdc45be5b584a). Great idea!